samples/generate-offline-map-with-custom-parameters/src/main/java/com/esri/arcgismaps/sample/generateofflinemapwithcustomparameters/screens/GenerateOfflineMapWithCustomParametersScreen.kt

```diff
@@ -0,0 +1,308 @@
+    fun setBottomSheetVisibility(isVisible: Boolean) {
+        showBottomSheet = isVisible
+    }
+
+    Scaffold(snackbarHost = { SnackbarHost(hostState = mapViewModel.snackbarHostState) },
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it)
+                    // Disable taps when job progress dialog is shown
+                    .clickable(enabled = !mapViewModel.showJobProgressDialog, onClick = { })
+            ) {
+                MapView(modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth()
+                    // Retrieve the size of the Composable MapView
+                    .onSizeChanged { size ->
+                        mapViewModel.updateMapViewSize(size)
+                    },
+                    arcGISMap = mapViewModel.arcGISMap,
+                    graphicsOverlays = listOf(mapViewModel.graphicsOverlay),
+                    mapViewInteractionOptions = MapViewInteractionOptions(isRotateEnabled = false),
+                    mapViewProxy = mapViewModel.mapViewProxy,
+                    onLayerViewStateChanged = {
+                        mapViewModel.calculateDownloadOfflineArea()
+                    },
+                    onViewpointChangedForCenterAndScale = {
+                        mapViewModel.calculateDownloadOfflineArea()
+                    }
+                )
+                // Show bottom sheet with override parameter options
+                if (showBottomSheet) {
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentSize(),
+                        onDismissRequest = { showBottomSheet = false },
+                        sheetState = sheetState
+                    ) {
+                        OverrideParameters(
+                            defineParameters = mapViewModel::defineGenerateOfflineMapParameters,
+                            setBottomSheetVisibility = ::setBottomSheetVisibility
+                        )
+                    }
+                }
+                // Display progress dialog while generating an offline map
+                if (mapViewModel.showJobProgressDialog) {
+                    JobLoadingDialog(title = "Generating offline map...",
+                        progress = mapViewModel.offlineMapJobProgress,
+                        cancelJobRequest = { mapViewModel.cancelOfflineMapJob() })
+                }
+
+                // Display a dialog if the sample encounters an error
+                mapViewModel.messageDialogVM.apply {
+                    if (dialogStatus) {
+                        MessageDialog(
+                            title = messageTitle, description = messageDescription, onDismissRequest = ::dismissDialog
+                        )
+                    }
+                }
```

Comment:
use of `apply` here is not necessary as you're not configuring the object and you're not using the result either. 
A simple if statement would do. 
```suggestion
                if (mapViewModel.messageDialogVM.dialogStatus) {
                    MessageDialog(
                            title = messageTitle, description = messageDescription, onDismissRequest = ::dismissDialog
                        )
                    }
                }
```

I would also suggest renaming `dialogStatus ` to describe what it is driving. Such as  `isDialogShown`, or `isDialogVisible` or `showDialog`

Reply:
This is a shared component in the sampleslib. So I'd be wary to change the names. And messageTitle, messageDescription and ::dismissDialog all come from the same component so need the apply block